src/AutorestSwift/View Models/ResponseViewModel.swift

```diff
@@ -42,13 +42,30 @@ enum ResponseBodyType: String {
     /// Service returns a byteArray body
     case byteArrayBody
 
-    static func strategy(for input: String, and type: AllSchemaTypes) -> ResponseBodyType {
+    case dateBody
+    case dateTimeBody
+    case datetimeRfc1123Body
+
+    static func strategy(for input: String, and type: AllSchemaTypes, schema: Schema) -> ResponseBodyType {
```

Comment:
If we now need more information than just the type enum, we should just make this:
```swift
static func strategy(for input: String, and schema: Schema) -> ResponseBodyType
```

Because type is a property of Schema.